extensions/Worker.Extensions.CosmosDB/src/Extensions/FunctionsWorkerApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Provides extension methods to work with a <see cref="IFunctionsWorkerApplicationBuilder"/>.
+    /// </summary>
+    public static class FunctionsWorkerApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the CosmosDB extension for FSharp functions.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The same instance of the <see cref="IFunctionsWorkerApplicationBuilder"/> for chaining.</returns>
+        public static IFunctionsWorkerApplicationBuilder ConfigureCosmosDB(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            if (builder is null)
+            {
+                throw new System.ArgumentNullException(nameof(builder));
+            }
+
+            // TODO: How to handle scenarios where this method is used by a C# function?
+            // Avoid double calling?
+            // Could we write an analyzer to detect this?
+
+            // Tested and this is working. Almost all F# projects have a hidden reference to the FSharp.Core assembly
+            if (!IsFSharp(Assembly.GetCallingAssembly()))
+            {
+                Console.WriteLine("FSharp not detected detected. Skipping CosmosDB extension configuration.");
+                return builder;
+                // Should we throw instead?
+                // throw new InvalidOperationException("The 'ConfigureCosmosDB' extension method is only for FSharp projects, please remove the call to 'ConfigureCosmosDB' from your startup code.");
+                // LOG: dotnet exited with code 134 (0x86). Unhandled exception. System.InvalidOperationException: The 'ConfigureCosmosDB' extension method is only for FSharp projects, please remove the call to 'ConfigureCosmosDB' from your startup code.
+            }
+
+            var cosmosdbStartup = new CosmosExtensionStartup();
```

Comment:
We can avoid instantiating this if we flip the implementation. The static extension method contains all the logic, and the startup class calls the extension method.